kong/status/ready.lua

```diff
@@ -111,6 +121,15 @@ return {
         ngx_log(ngx_NOTICE, "not ready for proxying: ", err)
         return kong.response.exit(503, { message = err })
       end
+    end,
+
+    POST = function(self, dao, helpers)
+      if self.params and self.params.status == "draining" then
+        kong_shm:set(KONG_STATUS_READY, false)
+        return kong.response.exit(204)
+      end
+
+      return kong.response.exit(501)
```

Comment:
```suggestion
      return kong.response.exit(400)
```
Feels like a 400 response is more appropriate